opentaskspal/src/main/java/org/dmfs/opentaskspal/contentpal/tables/Filtered.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.opentaskspal.contentpal.tables;
+
+import android.content.ContentProviderClient;
+import android.support.annotation.NonNull;
+
+import org.dmfs.android.contentpal.InsertOperation;
+import org.dmfs.android.contentpal.Operation;
+import org.dmfs.android.contentpal.Predicate;
+import org.dmfs.android.contentpal.Table;
+import org.dmfs.android.contentpal.UriParams;
+import org.dmfs.android.contentpal.View;
+import org.dmfs.android.contentpal.predicates.AllOf;
+
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public final class Filtered<T> implements Table<T>
```

Comment:
I'm not sure if a `Filtered` `Table` decorator is a good idea, because it changes the semantics. With the "scoped" tables we already have you can insert rows and you always update and delete the same rows and every `View` it returns will also return the rows you inserted.
That's not true for `Filtered` anymore. I can insert a row, but I can't update it if it doesn't pass the filter, nor can I read it from the `View` this table returns in that case.

Reply:
Ok, I've removed `Filtered` `Table` and `View`.